compiler/parser/RdrHsSyn.hs

```diff
@@ -2781,6 +2858,35 @@ mkTemplateDecls lname@(L nloc name) fields (L _ decls) = do
       }
     pureUnit = mkApp (mkUnqualVar $ mkVarOcc "pure") (noLoc $ ExplicitTuple noExt [] Boxed)
 
+-- Generate `newtype` and `instance` declarations corresponding to a
+-- `template instance InstanceName = T Arg1 .. ArgN`.
+mkTemplateInstance
+  :: Located RdrName              -- ^ Name given to template instance
+  -> LHsType GhcPs                -- ^ Application of generic template to type arguments
+  -> P (OrdList (LHsDecl GhcPs))  -- ^ Resulting declarations (`newtype` and `instance` of `TInstance` class)
+mkTemplateInstance instName@(L instLoc _) templateApp
+  | (templateType, tyArgs) <- splitHsAppTysPs templateApp
+  , L _ (HsTyVar NoExt NotPromoted templateName) <- templateType = do
+      let templateString = occNameString $ rdrNameOcc $ unLoc templateName
```

Comment:
```suggestion
      let templateString = getOccString templateName
```
This applies below as well since it works with located and not located stuff. There's also `getOccFS` which is faster and plays well with, say, `mkDataOccFS`.